### DISCUSSION TITLE
Ignore error for disappeared source files

### DISCUSSION THREAD

**Discussion by kalos** [creativeprojects/resticprofile#38] [General]
There is a way, in resticprofile or systemd service, to disable error on "Warning: failed to read all source data during backup"
https://github.com/restic/restic/issues/3098
My systemd service is:
```
[Service]
Type=notify
Restart=no
WorkingDirectory=/home/kalos/.config/resticprofile
Environment="HOME=/home/kalos"
ExecStart=/usr/local/bin/resticprofile --no-prio --no-ansi --config profiles.yaml --name all-my-backups backup
```
Thanks.

**Comment by creativeprojects** [comment]
Yes I can see how it would be an interesting feature.
The main problem is to detect this particular error:
- restic does not return different codes, it's one code for all errors and another when you stop the process (^C): https://github.com/restic/restic/issues/956
- I can pipe the output to analyse it, that is something I started to do for the status file and I suppose I could try to detect it there. I need a bit more testing for that

**Reply by creativeprojects** [reply]
No actually I was wrong, there's a different exit status **3** when a snapshot was created but some files were missing: https://restic.readthedocs.io/en/latest/040_backup.html#exit-status-codes 👍🏻

**Reply by kalos** [reply]
Sorry, maybe I haven't been exhaustive.

Since restic has a specific exit code, I think it's not very difficult to implement a check on it.
For this, I was asking you if you could get around the problem by working on the systemd service.

ps: Thank you so much for this very complete restic wrapper!

**Reply by creativeprojects** [reply]
Yes, considering restic is returning an exit code 3, it's easy to ignore it and make resticprofile return with a code 0 (telling systemd all was fine).

I'll add a parameter in the configuration for that though, I don't want to change the *default* behaviour 👍🏻

**Reply by creativeprojects** [reply]
It will be available in the next release (0.13.0)

https://github.com/creativeprojects/resticprofile#warnings-from-restic
